datafusion/physical-optimizer/Cargo.toml

```diff
@@ -32,9 +32,15 @@ rust-version = { workspace = true }
 workspace = true
 
 [dependencies]
+arrow = { workspace = true }
 arrow-schema = { workspace = true }
 datafusion-common = { workspace = true, default-features = true }
 datafusion-execution = { workspace = true }
+datafusion-expr-common = { workspace = true, default-features = true }
 datafusion-physical-expr = { workspace = true }
 datafusion-physical-plan = { workspace = true }
 itertools = { workspace = true }
+
+[dev-dependencies]
+datafusion-functions-aggregate = { workspace = true }
```

Comment:
I think we can't use function that hasn't implement accumulator here, since it is optimizer related to physical plan.

Error if I switch to stub
```
test aggregate_statistics::tests::test_count_partial_with_nulls_indirect_child ... FAILED
test aggregate_statistics::tests::test_count_partial_direct_child ... FAILED
test aggregate_statistics::tests::test_count_partial_with_nulls_direct_child ... FAILED
test aggregate_statistics::tests::test_count_partial_indirect_child ... FAILED
test aggregate_statistics::tests::test_count_inexact_stat ... FAILED
test aggregate_statistics::tests::test_count_with_nulls_inexact_stat ... FAILED

successes:

successes:

failures:

---- aggregate_statistics::tests::test_count_partial_with_nulls_indirect_child stdout ----
Error: NotImplemented("no impl for stub")

---- aggregate_statistics::tests::test_count_partial_direct_child stdout ----
Error: NotImplemented("no impl for stub")

---- aggregate_statistics::tests::test_count_partial_with_nulls_direct_child stdout ----
Error: NotImplemented("no impl for stub")

---- aggregate_statistics::tests::test_count_partial_indirect_child stdout ----
Error: NotImplemented("no impl for stub")

---- aggregate_statistics::tests::test_count_inexact_stat stdout ----
Error: NotImplemented("no impl for stub")

---- aggregate_statistics::tests::test_count_with_nulls_inexact_stat stdout ----
Error: NotImplemented("no impl for stub")
```